src/ashell/comms-shell.c

```diff
@@ -80,10 +80,6 @@ static void insert_char(char *pos, char c, uint8_t end)
 {
     char tmp;
 
-    /* Echo back to console */
-    if (echo_mode)
-        comms_write_buf(&c, 1);
```

Comment:
I still don't understand where there's another line in this insert_char function:
```
        comms_write_buf(&tmp, 1);
```

Can you make sure that's okay and maybe add some comment describing this strange function's purpose. :)